source/dem/velocity_verlet_integrator.cc

```diff
@@ -30,10 +31,14 @@ VelocityVerletIntegrator<dim>::integrate_half_step_location(
             g[d] +
             particle_force[d] / particle_properties[PropertiesIndex::mass];
 
+          // Half-step velocity
+          half_step_velocity[d] =
+            particle_properties[PropertiesIndex::v_x + d] +
+            0.5 * particle_acceleration[d] * dt;
+
           // Update particle position using half-step velocity
-          particle_position[d] +=
-            particle_properties[PropertiesIndex::v_x + d] * dt +
-            0.5 * dt * dt * particle_acceleration[d];
+          particle_position[d] += half_step_velocity[d] * dt +
+                                  0.5 * dt * dt * particle_acceleration[d];
```

Comment:
```suggestion
          particle_position[d] += half_step_velocity[d] * dt;
```

Reply:
This is the real formula for velocity verlet.
You also need to update the velocity itself. The half-step velocity has to be stored.